sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,38 +522,99 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      case s @ Sort(_, _, a: Aggregate) if a.resolved =>
```

Comment:
to be more clear, `ResolveSortReferences` tries to resolve attributes in `SortOrder`s, and `ResolveAggregateFunctions` tries to resolve aggregate functions in unexpected places(filter or sort), right?

So I think we should skip sort with aggregate functions here, i.e.
`case s: Sort if s.order.exists(ResolveAggregateFunctions.containsAggregate) => s`, and add comment to say this case should be handled in `ResolveAggregateFunctions`